### PR TITLE
Implementation of Bucket Sort in JS

### LINKED ISSUE #2124: Bucket Sort
- [x] C
- [x] C++
- [ ] C# - @luv4bytes
- [ ] CoffeeScript
- [ ] Dart
- [x] Go - @m-code12
- [x] Java
- [x] JS - @abhu-A-J
- [x] Kotlin - @shubhammantri1
- [ ] Php - @srimani-programmer
- [x] Python
- [x] Readme
- [x] Ruby - @m-code12
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by abhu-A-J**
Fixes #2124 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [ ] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Implementation of Bucket Sort in JS

#### Languages Used:
- JavaScript

#### Files Added:
-  Added Bucket_Sort.js file in Bucket_Sort folder.

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by abhu-A-J** [comment]
@raghav-dalmia Sir, can you review this in your free time. Thanks!

File: Bucket_Sort/Bucket_Sort.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/*
+ Bucket sort is a comparison sort algorithm that operates on elements by dividing them into different buckets and then sorting these buckets individually.
```
Comment (asha15) [inline Bucket_Sort/Bucket_Sort.js]: break this commend into two lines

File: Bucket_Sort/Bucket_Sort.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+*/
+
+// helper function to perform insertionSort
+
```
Comment (asha15) [inline Bucket_Sort/Bucket_Sort.js]: remove this line

File: Bucket_Sort/Bucket_Sort.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+// [ -11, -10, -1, 2, 4 ]
+console.log(bucketSort([-1, 2, 4, -10, -11]));
+
+// [ -11, -10, 4, 100, 200 ]
+console.log(bucketSort([100, 200, 4, -10, -11]));
+
+// [ -10, -4, -1, 11, 22 ]
+console.log(bucketSort([-1, 22, -4, -10, 11]));
+
```
Comment (asha15) [inline Bucket_Sort/Bucket_Sort.js]: remove this line

File: Bucket_Sort/Bucket_Sort.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
+/*
+ Bucket sort is a comparison sort algorithm that operates on elements by dividing 
+ them into different buckets and then sorting these buckets individually.
+*/
```
Comment (asha15) [inline Bucket_Sort/Bucket_Sort.js]: keep a line space after this

File: Bucket_Sort/Bucket_Sort.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                arr[j + 1] = arr[j];
+            }
+        }
+        arr[j + 1] = currentVal;
+    }
+    return arr;
+}
+
+// function to perform bucket sort
+
```
Comment (asha15) [inline Bucket_Sort/Bucket_Sort.js]: remove this line

File: Bucket_Sort/Bucket_Sort.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+// I/P and O/P examples
+
+// [ -11, -10, -1, 2, 4 ]
+console.log(bucketSort([-1, 2, 4, -10, -11]));
+
+// [ -11, -10, 4, 100, 200 ]
+console.log(bucketSort([100, 200, 4, -10, -11]));
+
+
+
```
Comment (asha15) [inline Bucket_Sort/Bucket_Sort.js]: remove this three lines

**Comment by abhu-A-J** [comment]
@asha15 Ma'am done with the changes. Can you review again!

**Comment by asha15** [comment]
keep going :)

**Pull request merged**